linera-core/src/remote_node.rs

```diff
@@ -256,17 +260,17 @@ impl<N: ValidatorNode> RemoteNode<N> {
 
     /// Downloads the blobs with the given IDs. This is done in one concurrent task per block.
     /// Each task goes through the validators sequentially in random order and tries to download
-    /// it. Returns all the blobs it could find, and silently ignores the ones it could not.
+    /// it. Returns `None` if it couldn't find all blobs.
     #[instrument(level = "trace", skip(validators))]
-    pub async fn download_blobs(blob_ids: &[BlobId], validators: &[Self]) -> Vec<Blob> {
-        future::join_all(
-            blob_ids
-                .iter()
-                .map(|blob_id| Self::download_blob(validators, *blob_id)),
-        )
-        .await
-        .into_iter()
-        .flatten()
-        .collect::<Vec<_>>()
+    pub async fn download_blobs(blob_ids: &[BlobId], validators: &[Self]) -> Option<Vec<Blob>> {
+        let mut stream = blob_ids
+            .iter()
+            .map(|blob_id| Self::download_blob(validators, *blob_id))
+            .collect::<FuturesUnordered<_>>();
+        let mut blobs = Vec::new();
+        while let Some(maybe_blob) = stream.next().await {
+            blobs.push(maybe_blob?);
+        }
```

Comment:
Similar question - does this cancel? do we care?

Reply:
Yes, it cancels, and if I understand correctly, we don't care:
In all the call sites we ended up comparing the length of the returned blob list with the length of the requested blob IDs list. So why not return immediately when we know we didn't get all the blobs?